mitosheet/src/components/taskpanes/DataframeImport/DataframeImportTaskpane.tsx

```diff
@@ -35,7 +58,13 @@ const DataframeImportTaskpane = (props: DataframeImportTaskpaneProps): JSX.Eleme
     const [dfNamesInNotebook, setDfNamesInNotebook] = useState<string[]>([]);
 
     const {params, setParams, edit} = useSendEditOnClick<{df_names: string[]}, {df_names: string[]}>(
-        {df_names: []}, StepType.DataframeImport, props.mitoAPI, props.analysisData, {allowSameParamsToReapplyTwice: true}
+        {df_names: []}, 
+        StepType.DataframeImport, 
+        props.mitoAPI, 
+        props.analysisData, 
+        {allowSameParamsToReapplyTwice: true}, 
+        // If we're updating already imported data, then pass the override edit function
+        props.updateImportedData === undefined ? undefined : (newImportedData) => updateImportedDataWithDataframe(props.updateImportedData, newImportedData, props.setUIState)
```

Comment:
Oh. I see what you did here. Hm. Let me think on this one.

Reply:
There's another way to factor this, that at first blush seems much more maintainable to me long-term:
1. Abstract the DataframeImportTaskpane into a DataframeImportScreen and DataframeImportTaskpane. 
2. DataframeImportScreen is just pretty much everything except `useSendEditOnClick`. It takes params, setParams, and edit() function as props.
3. DataframeImportTaskpane is a thin wrapped on top of this, that has useSendEditOnClick and passes these directly to the DataframeImportScreen
4. You consume the DataframeImportScreen directly in the UpdateImports taskpane, without needing to worry about this function.

The benefits of this is that you don't have to change the `useSendEditOnClick` function, which is already quite complex, and probably should only be responsible for sending edits to the backend. This also doesn't really require any code duplication, as well, so it's not a messy refactor. It also ends with different components being responsible for different things in a way that is very sensible and not hard to reason about.